src/libmongoc/src/mongoc/mongoc-error.c

```diff
@@ -77,10 +77,6 @@ _mongoc_write_error_is_retryable (bson_error_t *error)
    case MONGOC_SERVER_ERR_NOTMASTERORSECONDARY:
       return true;
    default:
-      if (strstr (error->message, "not master") ||
```

Comment:
Related to a comment Jeremy had made in the review for CDRIVER-3615: https://github.com/mongodb/mongo-c-driver/pull/601/files#r421624638

It may be worth checking that `error->domain == MONGOC_ERROR_SERVER` or `error->domain == MONGOC_ERROR_WRITE_CONCERN` in this function to assure that this is an error that originated from the server. This function should only be called with an error parsed using MONGOC_ERROR_API_VERSION_2, so checking those domains should be all that is necessary.